hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/block/TestDeletedBlockLog.java

```diff
@@ -85,23 +86,24 @@
   private ContainerManagerV2 containerManager;
   private StorageContainerManager scm;
   private List<DatanodeDetails> dnList;
-  private DBTransactionBuffer dbTransactionBuffer;
+  private SCMHADBTransactionBuffer SCMHADBTransactionBuffer;
 
   @Before
   public void setup() throws Exception {
     testDir = GenericTestUtils.getTestDir(
         TestDeletedBlockLog.class.getSimpleName());
     conf = new OzoneConfiguration();
+    conf.setBoolean(ScmConfigKeys.OZONE_SCM_HA_ENABLE_KEY, true);
     conf.setInt(OZONE_SCM_BLOCK_DELETION_MAX_RETRY, 20);
     conf.set(HddsConfigKeys.OZONE_METADATA_DIRS, testDir.getAbsolutePath());
     scm = TestUtils.getScm(conf);
     containerManager = Mockito.mock(ContainerManagerV2.class);
-    dbTransactionBuffer =
-        new MockDBTransactionBuffer(scm.getScmMetadataStore().getStore());
+    SCMHADBTransactionBuffer =
+        new MockSCMHADBTransactionBuffer(scm.getScmMetadataStore().getStore());
```

Comment:
Is it possible to implement "flush on every write" in `MockSCMHADBTransactionBuffer` so in tests we no longer need to call flush()?

Reply:
Or, even better, `MockSCMHADBTransactionBuffer` just pass through the buffer and write to table directly?

Reply:
Agree. Can `MockSCMHADBTransactionBuffer` share the same logic `addToBuffer/removeFromBuffer` as `SCMDBTransactionBufferImpl`, and let its `flush` to throw exception ?

Reply:
I think, MockInstances should mimic every functional behaviour of the class. I would prefer to keep it as it is for now. We should be able to test the batch behaviour of SCMHADBTransaction behaviour with this.

Reply:
I see. Sure it sounds good.

Reply:
One tricky thing is that, recon uses `MockSCMHAManager` and `MockSCMHADBTransactionBuffer`.
Since Recon will not use ratis, `flush` is not called, thus there will finally be OOM for recon.

```
Ctor of ReconStorageContainerManagerFacade
    this.scmhaManager = MockSCMHAManager.getInstance(
        true, new MockSCMHADBTransactionBuffer(dbStore));

```